salt/docker/init.sls

```diff
@@ -83,13 +83,20 @@ docker:
     - onlyif: systemctl status docker.service
     - onchanges:
       - /etc/systemd/system/docker.service.d/proxy.conf
-      - /etc/docker/daemon.json
     - require:
       - file: /etc/sysconfig/docker
-      - file: /etc/docker/daemon.json
   service.running:
     - enable: True
     - watch:
       - pkg: docker
       - file: /etc/sysconfig/docker
       - /etc/systemd/system/docker.service.d/proxy.conf
+
+docker-reload-config:
+  cmd.run:
+    - name: systemctl reload docker.service
+    - onlyif: systemctl status docker.service
+    - onchanges:
+      - file: /etc/docker/daemon.json
+    - require:
```

Comment:
The `require` here means that the `docker:` state will be executed and, on success, the `docker-reload-config` will be executed too, and that is not what we want: we want to do the reload _iff_ `/etc/docker/daemon.json` changes. So I would remove the `require` and only leave the `onchanges`

Reply:
@inercia fixed, thanks